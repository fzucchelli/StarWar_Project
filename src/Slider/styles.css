.content {
  vertical-align: middle;
  display: inline-block;
  margin-left: 26px;
  text-align: center;
  width: 100%;
  white-space: nowrap;
  overflow-y: hidden;
}
.card {
  background-color: #cc5252;
  opacity: 0.93;
  border-radius: 20px;
  display: inline-block;
  height: 520px;
  margin: 26px;
  margin-right: 26px;
  margin-top: 15%;
}
.card .card-content {
  position: absolute;
  padding: 0;
  padding-bottom: 10px;
  bottom: 0;
}

.title {
  font-size: 20px;
  line-height: 1;
  color: #f7f5f5;
  font-weight: bold;
  text-transform: capitalize;
  white-space: initial;
}
.subTitle {
  font-size: 11.5px;
  color: #070707;
  font-weight: bold;
  text-transform: uppercase;
  white-space: initial;
}
.para {
  padding-left: 19px;
  padding-right: 19px;
  line-height: 1.583;
  text-align: left;
  font-size: 1rem;
  color: #3a3a3a;
  white-space: initial;
  margin-bottom: 19px;
  box-shadow: 0 0 black;
}
.icon {
  color: grey;
  padding-top: 0;
  padding-left: 19px;
  padding-bottom: 19px;
  box-shadow: none;
  background: none;
}
.icon:hover {
  background: none;
  box-shadow: none;
}
.activator {
  height: 160px;
  width: 100%;
}
img.circle {
  border-radius: 50% !important;
  height: 60px !important;
  width: 60px !important;
}
.row .col.s3 {
  margin-left: 0;
  padding: 0;
}
.header {
  margin: 19px;
}
.icon-btn {
  text-align: center;
}
.icon-btn button {
  background: none;
  border: 0;
}

.card-image {
  height: 150px;
}

img.circle-film {
  border-radius: 50% !important;
  height: 30px !important;
  width: 30px !important;
}

body {
  background-color: #d8d3d3;
  background-image: url("/Star-Wars-Wallpaper.jpeg");

  /* Full height */
  height: 100%;

  /* Center and scale the image nicely */
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
}

body,
html {
  height: 100%;
}

@media (max-width: 960px) {
  .card {
    float: left;
    clear: right;
  }
}
