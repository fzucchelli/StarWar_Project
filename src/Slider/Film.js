import React from "react";
import "./styles.css";

export default class Film extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      filmTitle: "",
      release_date: ""
    };
  }

  componentDidMount() {
    //
    fetch(this.props.filmTitle)
      .then(response => response.json())
      .then(data => {
        //console.log(data);
        this.setState({
          filmTitle: data.title,
          release_date: data.release_date
        });
      });
  }

  render() {
    return (
      <div className="row valign-wrapper header">
        <div className="col s3">
          <img
            src="star-wars.png"
            alt=""
            className="circle-film responsive-img"
          />
        </div>
        <div className="col s10">
          <span className="subTitle">
            {this.state.filmTitle}
            {"  "}
            {`(${this.state.release_date})`}
          </span>
        </div>
      </div>
    );
  }
}
