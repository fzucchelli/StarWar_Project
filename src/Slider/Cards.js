import React from "react";
import Film from "./Film";
import "./styles.css";

class Cards extends React.Component {
  ColorChange = e => {
    var heart = e.currentTarget;
    if (heart.style.color !== "green") {
      heart.style.color = "green";
    } else {
      heart.style.color = "grey";
    }
  };
  render() {
    return (
      <div className="content">
        {this.props.cardsToShow.map((data, i) => {
          return (
            <div
              className="card"
              id="card"
              key={data.id}
              style={this.props.cardStyle}
            >
              <div className="card-image waves-effect waves-block waves-light">
                <a href={data.href}>
                  <div className="row valign-wrapper header">
                    <div className="col s3">
                      <img
                        src="images.png"
                        alt=""
                        className="circle responsive-img"
                      />
                    </div>
                    <div className="col s10">
                      <span className="title">{data.name}</span>
                      <br />
                      <span className="subTitle">{data.model}</span>
                      <br />
                      <span className="subTitle">Crew: {data.crew}</span>
                    </div>
                  </div>
                </a>
              </div>
              <hr />
              <p className="para">
                <b>Films</b>
              </p>
              <div className="card-content" />
              {data.films.map(film => (
                <p>
                  <Film filmTitle={film} />
                </p>
              ))}
            </div>
          );
        })}
      </div>
    );
  }
}
export default Cards;
