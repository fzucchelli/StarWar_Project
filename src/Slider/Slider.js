import React from "react";
import Cards from "./Cards";
import { CARDS_API } from "../util/config";
import PropTypes from "prop-types";

class Slider extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      cardData: [],
      currentCard: 0,
      position: 0,
      cardStyle: {
        transform: "translateX(0px)",
        width: "0px"
      },
      width: 0
    };
  }

  componentDidMount() {
    //
    fetch(CARDS_API)
      .then(response => response.json())
      .then(data => {
        const width = 924 / this.props.numCardsToShow;
        const cardStyle = {
          transform: "translateX(0px)",
          width: `${width}px`
        };
        console.log(data.results);
        this.setState({
          cardData: data.results.filter(ship => ship.crew > 10),
          cardStyle: cardStyle,
          width: width
        });
      });
  }

  // func: click the slider buttons
  handleClick(type) {
    // get the card's margin-right
    let margin = window.getComputedStyle(document.getElementById("card"))
      .marginLeft;
    margin = JSON.parse(margin.replace(/px/i, ""));

    const cardWidth = this.state.width; // the card's width
    const cardMargin = margin; // the card's margin
    const cardNumber = this.state.cardData.length; // the number of cards
    let currentCard = this.state.currentCard; // the index of the current card
    let position = this.state.position; // the position of the cards

    // slide cards
    if (
      type === "next" &&
      currentCard < cardNumber - this.props.numCardsToShow
    ) {
      currentCard++;
      position -= cardWidth + cardMargin;
    } else if (type === "prev" && currentCard > 0) {
      currentCard--;
      position += cardWidth + cardMargin;
    }
    this.setCard(currentCard, position);
  }

  setCard(currentCard, position) {
    this.setState({
      currentCard: currentCard,
      position: position,
      cardStyle: {
        transform: `translateX(${position}px)`,
        transition: "transform .3s ease-out",
        width: "308px"
      }
    });
  }

  render() {
    const leftArrowStyle = {
      color: ` ${this.state.currentCard === 0 ? "grey" : "green"}`
    };
    const rigthArrowStyle = {
      color: ` ${
        this.state.currentCard ===
        this.state.cardData.length - this.props.numCardsToShow
          ? "grey"
          : "green"
      }`
    };
    return (
      <div>
        <Cards
          cardsToShow={this.state.cardData}
          cardStyle={this.state.cardStyle}
        />
        <div className="icon-btn">
          <button
            style={leftArrowStyle}
            onClick={() => this.handleClick("prev")}
          >
            <i className="medium material-icons">chevron_left</i>
          </button>
          <button
            style={rigthArrowStyle}
            onClick={() => this.handleClick("next")}
          >
            <i className="medium material-icons">chevron_right</i>
          </button>
        </div>
      </div>
    );
  }
}

Slider.defaultProps = {
  numCardsToShow: 3
};

Slider.propTypes = {
  numCardsToShow: PropTypes.number
};

export default Slider;
