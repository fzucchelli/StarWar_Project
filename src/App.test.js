import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";
import Slider from "./Slider/Slider";

import { shallow } from "enzyme";

test("should render Slider correctly", () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<Slider />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

/* test("should call handleClick on button click", () => {
  const handleClick = jest.fn();
  const wrapper = shallow(<Slider numCardsToShow={3} />);
  wrapper.find("button").simulate("click");
  expect(handleClick).toHaveBeenCalled();
}); */

/* test('Should have proper border-top, border-right & background-color when disabled', () => {
  const renderedComponent = shallow(<LeftArrow disabled />);
  expect(renderedComponent).toHaveStyleRule('background-color', '#B2B2B2');
  expect(renderedComponent).toHaveStyleRule('border-top', 'solid 2px #B2B2B2', {
    modifier: ':before',
  });
  expect(renderedComponent).toHaveStyleRule('border-right', 'solid 2px #B2B2B2', {
    modifier: ':before',
  });
}); */
