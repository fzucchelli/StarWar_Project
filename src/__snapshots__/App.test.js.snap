// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Slider correctly 1`] = `
<div>
  <Cards
    cardStyle={
      Object {
        "transform": "translateX(0px)",
        "width": "0px",
      }
    }
    cardsToShow={Array []}
  />
  <div
    className="icon-btn"
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "color": " grey",
        }
      }
    >
      <i
        className="medium material-icons"
      >
        chevron_left
      </i>
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "color": " green",
        }
      }
    >
      <i
        className="medium material-icons"
      >
        chevron_right
      </i>
    </button>
  </div>
</div>
`;
